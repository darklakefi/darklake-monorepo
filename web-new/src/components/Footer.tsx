import { socials } from "@/constants/links";
import Image from "next/image";

const Footer = () => {
  return (
    <footer className="flex flex-row justify-between items-center pb-[80px]">
      <button className="uppercase text-brand-30 underline active-hover-opacity">What is MEV?</button>
      <div className="flex flex-row gap-x-[40px]">
        {socials.map((social) => (
          <a href={social.url} key={social.title} target="_blank" rel="noreferrer" className="active-hover-opacity">
            <Image src={social.iconUrl} alt={social.title} height={24} width={24} />
          </a>
        ))}
      </div>
    </footer>
  );
};

export default Footer;
