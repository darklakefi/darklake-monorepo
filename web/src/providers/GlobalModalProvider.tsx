"use client";

import React, { createContext, useMemo, useState } from "react";
import WhatIsMevModal from "@/components/Modal/WhatIsMevModal";

export enum GlobalModalType {
  WHAT_IS_MEV = "WHAT_IS_MEV",
}

interface IGlobalModalContext {
  openModal: (type: GlobalModalType) => void;
}

export const GlobalModalContext = createContext<IGlobalModalContext | null>(null);

export default function GlobalModalProvider({ children }: { children: React.ReactNode }) {
  const [visibleModalType, setVisibleModalType] = useState<GlobalModalType | undefined>(undefined);

  const context = useMemo(
    () => ({
      openModal: (type: GlobalModalType) => setVisibleModalType(type),
    }),
    [],
  );

  return (
    <GlobalModalContext.Provider value={context}>
      {children}
      <WhatIsMevModal
        isOpen={visibleModalType === GlobalModalType.WHAT_IS_MEV}
        onClose={() => setVisibleModalType(undefined)}
      />
    </GlobalModalContext.Provider>
  );
}
