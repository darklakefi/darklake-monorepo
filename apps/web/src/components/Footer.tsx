"use client";

import { socials, SocialType } from "@/constants/links";
import useGlobalModal from "@/hooks/useGlobalModal";
import { GlobalModalType } from "@/providers/GlobalModalProvider";
import { cn } from "@/utils/common";

const Footer = () => {
  const { openModal } = useGlobalModal();

  return (
    <footer className="flex flex-col sm:flex-row justify-between items-center z-20 py-20">
      <div className="flex items-center justify-between gap-x-10 max-sm:mb-10">
        <button
          className="uppercase underline text-brand-30 text-lg leading-6 hover-with-active"
          onClick={() => openModal(GlobalModalType.WHAT_IS_MEV)}
        >
          What is MEV?
        </button>
        <a
          href={process.env.NEXT_PUBLIC_REPORT_BUG_URL}
          title="Report a bug"
          target="_blank"
          className="uppercase underline text-brand-30 text-lg leading-6 hover-with-active"
        >
          Report a bug
        </a>
      </div>
      <div className="flex flex-row items-center gap-x-[40px]">
        {socials.map((social) => (
          <a href={social.url} key={social.title} target="_blank" rel="noreferrer" className="hover-with-active">
            <i
              className={cn(
                social.iconClassName,
                social.type !== SocialType.TELEGRAM && "text-brand-30 text-xl",
                social.type === SocialType.TELEGRAM && "size-5",
              )}
            />
          </a>
        ))}
      </div>
    </footer>
  );
};

export default Footer;
