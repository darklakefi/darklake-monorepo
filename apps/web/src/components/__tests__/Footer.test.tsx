import { act, render, RenderResult } from "@testing-library/react";
import Footer from "@/components/Footer";
import GlobalModalProvider from "@/providers/GlobalModalProvider";

describe("<Footer />", () => {
  it("renders successfully", async () => {
    let baseElement: RenderResult["baseElement"] | undefined;

    await act(async () => {
      ({ baseElement } = render(
        <GlobalModalProvider>
          <Footer />
        </GlobalModalProvider>,
      ));
    });

    expect(baseElement).toMatchSnapshot();
  });
});
