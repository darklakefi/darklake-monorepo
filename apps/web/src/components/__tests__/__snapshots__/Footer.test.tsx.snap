// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders successfully 1`] = `
<body>
  <div>
    <footer
      class="flex flex-col sm:flex-row justify-between items-center z-20 py-20"
    >
      <div
        class="flex items-center justify-between gap-x-10 max-sm:mb-10"
      >
        <button
          class="uppercase underline text-brand-30 text-lg leading-6 tracking-normal hover-with-active"
        >
          What is MEV?
        </button>
        <a
          class="uppercase underline text-brand-30 text-lg leading-6 tracking-normal hover-with-active"
          target="_blank"
          title="Report a bug"
        >
          Report a bug
        </a>
        <a
          class="uppercase underline text-brand-30 text-lg leading-6 tracking-normal hover-with-active"
          href=""
          target="_blank"
          title="Blog"
        >
          Blog
        </a>
      </div>
      <div
        class="flex flex-row items-center gap-x-[40px]"
      >
        <a
          class="hover-with-active"
          href="https://x.com/darklakefi"
          rel="noreferrer"
          target="_blank"
        >
          <i
            class="hn hn-x text-brand-30 text-xl"
          />
        </a>
        <a
          class="hover-with-active"
          href="https://github.com/darklakefi"
          rel="noreferrer"
          target="_blank"
        >
          <i
            class="hn hn-github text-brand-30 text-xl"
          />
        </a>
        <a
          class="hover-with-active"
          href="https://t.me/darklakeficonvo"
          rel="noreferrer"
          target="_blank"
        >
          <i
            class="hn hn-telegram size-5"
          />
        </a>
      </div>
    </footer>
  </div>
</body>
`;
